src/connector/src/source/filesystem/opendal_source/opendal_enumerator.rs

```diff
@@ -51,14 +51,22 @@ impl<Src: OpendalSource> SplitEnumerator for OpendalEnumerator<Src> {
 
     async fn list_splits(&mut self) -> ConnectorResult<Vec<OpendalFsSplit<Src>>> {
         let empty_split: OpendalFsSplit<Src> = OpendalFsSplit::empty_split();
+        let prefix = self.prefix.as_deref().unwrap_or("/");
 
-        Ok(vec![empty_split])
+        match self.op.list(prefix).await {
+            Ok(_) => return Ok(vec![empty_split]),
+            Err(e) => {
+                return Err(e
+                    .with_context("", "fail to create source, please check your config")
```

Comment:
It seems that you are using the `with_context` from [opendal](https://opendal.apache.org/docs/rust/opendal/struct.Error.html#method.with_context) to push the context string as part of the opendal error. I think it is inappropriate because opendal doesn't have the "create source" concept.

I thought @xiangjinwu means we should use the `context` method from anyhow to populate the context string. Example: https://github.com/risingwavelabs/risingwave/blob/40ccdd490b6d9d7a9481e429461bc8d4f6a6f5bd/src/batch/src/executor/insert.rs#L239